donkeycar/pipeline/sequence.py

```diff
@@ -1,7 +1,7 @@
-from typing import Any, Callable, Generic, Iterable, Iterator, List, Sized, Tuple, TypeVar
+from typing import Any, Callable, Generic, Iterable, Iterator, List, Sized, \
```

Comment:
Is it possible to leave the `sequence` as before ? The reason I ask is the _only_ important change here is to return a new `iterator` instance when a `__iter__(self)` is called. There are a lot more changes here. 

Even though some of your changes are good, I want to be a bit more thoughtful about them. If possible, lets leave the old shallow `list()` approach that allows training, and make fewer changes here.

Reply:
I have intentionally changed this because the list is iterable and will always return a new iterator which is standard python, whereas an iterator only returns self. This means we want to work w/ iterables, the iterator is just a small pointer. If we want to work with iterators then I need to implement `__iter__` to not return self for the iterator which I can do, but this is not very pythonic I think. 
Have a look at the whole file and not only at the change as it looks much more understandable now, because you only work with the shallow lists and never with the iterators directly.  
I will do what you suggest alternatively and push that, so you can compare.

Reply:
Done that, but pls have a look, the Iterators now have an Iterable-like `__iter__` which is a bit strange...